node/subsystem/src/messages.rs

```diff
@@ -25,11 +25,28 @@
 use futures::channel::{mpsc, oneshot};
 
 use polkadot_primitives::v1::{
-	BlockNumber, Hash, CommittedCandidateReceipt,
-	CandidateReceipt, PoV, ErasureChunk, BackedCandidate, Id as ParaId,
-	SignedAvailabilityBitfield, SigningContext, ValidatorId, ValidationCode, ValidatorIndex,
-	CoreAssignment, CoreOccupied, HeadData, CandidateDescriptor,
-	ValidatorSignature, OmittedValidationData, AvailableData,
+	AvailableData,
+	BackedCandidate,
+	BlockNumber,
+	CandidateDescriptor,
+	CandidateEvent,
+	CandidateReceipt,
+	CommittedCandidateReceipt,
+	CoreAssignment,
+	CoreOccupied,
+	CoreState,
+	ErasureChunk,
+	Hash,
+	HeadData,
+	Id as ParaId,
+	OmittedValidationData,
+	PoV,
+	SignedAvailabilityBitfield,
+	SigningContext,
+	ValidationCode,
+	ValidatorId,
+	ValidatorIndex,
+	ValidatorSignature,
```

Comment:
Not sure if thats ok; if that's the IDE sorting imports (rust-analyzer can do that I think) could that at least not do the one-per-line thing?

Reply:
I did that intentionally: I was having trouble diffing the imports when resolving a merge conflict in which people added and removed various items in different places. Using one per line, alphabetically, made it much easier to sort things out.

Reply:
It would be better to have them be more compact. We don't use this style anywhere else.

Reply:
As a question of policy, I'd prefer to leave them like this, and expand the use of this style. Rationale: the vast majority of the time, we're not looking at the import lines, but elsewhere in the file. Of the times we are in fact looking at the import lines, it's when adding or merging imports; in those times, it's faster when the lists are broad and alphabetical than when using basically any other taxonomy.

Nevertheless, 8e5c6da.

Reply:
Maybe the general rule should be to merge all crates import into one import, after that the compiler will ensure that you don't have duplicates in this list.